.github/workflows/cd.yaml

```diff
@@ -0,0 +1,59 @@
+name: CD (Docker Build & Push)
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      FULL_IMAGE_TAG: ${{ steps.tag.outputs.tag }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set Image Tag
+        id: tag
+        run: |
+          export GIT_TAG=$(git describe --abbrev=7 --always --tags)
+          echo "::set-output name=tag::stg-${GIT_TAG}"
+
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    needs: prepare
+    steps:
+      - name: Echo tag
+        id: echotag
+        env:
+          IMAGE_TAG: ${{ needs.prepare.outputs.FULL_IMAGE_TAG }}
```

Comment:
Isn't the output above just called `tag` rather than `FULL_IMAGE_TAG`? (Note that I've never written a GitHub Actions workflow, and I'm more or less just guessing what this syntax means, so I'm probably totally wrong.)

Reply:
yeah, let me dig into what i was thinking here, looks like copypasta though

Reply:
i think I missed some extra steps here, pushing up a fix. Thanks for catching this.

Reply:
ha, line 13 in this file is why this works (e.g. using FULL_IMAGE_TAG & not tag)

Reply:
just triple checked to make sure it was lined up. the reason i used FULL_IMAGE_TAG for the outputs instead of just tag was to make it super clear what variable is getting passed around, and tag by itself didn't seem descriptive enough.